csaf_2.0/json_schema/csaf_json_schema.json

```diff
@@ -154,13 +154,127 @@
+              "minLength": 5
+            },
+            "hashes": {
+              "$comment": "TODO: Array of objects (algorithm, value, optional name)",
+              "title": "List of hashes",
+              "description": "Contains a list of cryptographic hashes usable to identify files.",
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "title": "Cryptographic hash",
+                "description": "Contains all information to identify a file based on its cryptographic hash value.",
+                "type": "object",
+                "required": [
+                  "algorithm",
+                  "file",
+                  "value"
+                ],
+                "properties": {
+                  "algorithm": {
+                    "title": "Algorithm of the cryptographic hash",
+                    "description": "Contains the name of the cryptographic hash algorithm used to calculate the value.",
+                    "type": "string",
+                    "enum": [
+                      "SHA-256",
+                      "SHA-384",
+                      "SHA-512",
+                      "SHA-3",
+                      "BLAKE3"
+                    ]
+                  },
+                  "file": {
+                    "title": "Filename",
+                    "description": "Contains the filename of the file which is identified by the hash value.",
+                    "type": "string",
+                    "minLength": 1,
+                    "examples": [
+                      "WINWORD.EXE",
+                      "msotadddin.dll",
+                      "sudoers.so"
+                    ]
+                  },
+                  "value": {
+                    "title": "Value of the cryptographic hash",
+                    "description": "Contains the cryptographic hash value.",
+                    "type": "string",
+                    "minLength": 64,
+                    "pattern": "^([0-9a-fA-F]){64,}$",
+                    "examples": [
+                      "4775203615d9534a8bfca96a93dc8b461a489f69124a130d786b42204f3341cc",
+                      "9ea4c8200113d49d26505da0e02e2f49055dc078d1ad7a419b32e291c7afebbb84badfbd46dec42883bea0b2a1fa697c",
+                      "37df33cb7464da5c7f077f4d56a32bc84987ec1d85b234537c1c1a4d4fc8d09dc29e2e762cb5203677bf849a2855a0283710f1f5fe1d6ce8d5ac85c645d0fcb3"
+                    ]
+                  }
+                }
+              }
+
+            },
+            "purl": {
+              "$comment": "TODO: This should have a full regular expression to enforce purl syntax.",
```

Comment:
Purls are valid URLs so any regex we include here will be a generic URL regex with the restriction on the protocol part, i.e. `^pkg:`. I probably wouldn't bother with a URL regex and leave it as is, i.e. `"pattern": "^pkg:",`.

Reply:
Me neither - as indicated by my comment - reading the README.md spec of Steve I do not feel confident to write a terse and useful regex. The parsers of these documents will do this anyway  they like (and with security aspects implemented) - our patterns are more like shapes (my view). Not a copy-paste template for implementations.